### DISCUSSION TITLE
"Hired hand" for item transport (Sub/Outpost)

### DISCUSSION THREAD

**Discussion by James-E-A** [FakeFishGames/Barotrauma#8935] [Ideas and suggestions]
Say you're on a ship without a fabrication room, and you've got 12 empty ammo crates and 17 empty metal crates (from purchases) sitting in the docking port. If you have a moderate-sized (human) crew that's willing to help you haul it all up, it can go quickly… but that's not always the case.

It would be nice if you could find someone at an Outpost who would haul items to/from your sub. Talking to them pulls up an "item scanner" interface that comes with a few selectable filters:

* Search based on "Room" (linked hulls on the sub) or "Outpost" which filters for non-redhand items
* Filter out non-empty containers
* Filter out containers with less/more than Xmk contained item value
* Filter out ammo boxes with greater than X% remaining

Upon confirming the haul, the items would all be shortly dropped at the feet of the summoning player, or sent to the destination hull in the sub.

A small fee per-item—maybe 1mk for stacks or one-handers and 2mk for two-handers—would keep this balanced in singleplayer and give the crew an incentive to help out anyway in multiplayer, and could even offer roleplay opportunities (say a cranky engineer "has to slip a twenty to the locals because his goodfernuthin' crew can't find the thorium ore if it gave 'em cancer from staring at it").

**Comment by James-E-A** [comment]
This could probably be _easily_ generalized into a cross-sub item transport service, as well.

**Comment by James-E-A** [comment]
This could also be a use for reputation:

* Increased fees below +10 rep
* _Small_ (but accelerating) chance of items going missing starting at −0 rep; the service becomes realistically useless below −15

**Comment by mLuby** [comment]
The [mid-round purchases](https://github.com/Regalis11/Barotrauma/discussions/8970) suggestion could work nicely with this idea: when you exit the vendor interface, the items (in crates if >8 items) spawn by the vendor's feet. You can then carry the stuff to your sub yourself, or you can pay station dwellers to carry the stuff on the station's floor to your sub's cargo spawners (ignoring stolen stuff, of course). If the movers get stuck for a while, they just drop what they're carrying and leave.

Bonus: pay to have station dwellers carry all items (that they can access) from your current sub to the station floor and drop them. Then you can swap to a different sub and pay again to have all items on the station floor delivered to the sub's cargo spawners. Aside from being more convenient it'd add significant immersion. Also opens the door for events like "someone stole something during the move" and station reputation-related things like "stations will only help you move if rep > 50." (Might even try to steal from you or your sub if reputation < 50! Post guards.)
